go/host/host.go

```diff
@@ -891,45 +900,30 @@ func (h *host) bootstrapHost() types.Block {
 	return *currentBlock
 }
 
-func (h *host) awaitSecret() error {
-	// start listening for l1 blocks that contain the response to the request
-	listener, subs := h.ethClient.BlockListener()
+func (h *host) awaitSecret(fromHeight *big.Int) error {
+	blkStream, err := h.blockProvider.StartStreamingFromHeight(fromHeight)
+	if err != nil {
+		return err
+	}
 
 	for {
 		select {
-		case err := <-subs.Err():
-			h.logger.Error("Restarting L1 block monitoring while awaiting for secret. Errored with: %s", log.ErrKey, err)
-			// todo this is a very simple way of reconnecting the host, it might need catching up logic
-			listener, subs = h.ethClient.BlockListener()
+		case err := <-h.blockProvider.Err():
+			h.logger.Crit("blockProvider error", log.ErrKey, err)
```

Comment:
Related to the reconnect question from the interface definition.
is it desired to kill the host here?

Reply:
My thinking was blockprovider handlwe reconnections internally and only use the error channel if it's completely broken, like can't connect to L1 for 10mins or something.

But actually let's scrap that, the host will notice if it hasn't received a block for X amount of time and can handle that itself, its components should be resilient so the lifecycle can be controlled from one place.